pkg/process/checks/format.go

```diff
@@ -0,0 +1,299 @@
+		data.CPUCount = len(proc.Info.Cpus)
+		data.Memory = uint64(proc.Info.TotalMemory)
+		for _, p := range proc.Processes {
+			processes[p.Pid] = p
+			pids = append(pids, int(p.Pid))
+		}
+
+		for _, c := range proc.Containers {
+			containers[c.Id] = c
+		}
+	}
+
+	for cid := range containers {
+		containerIDs = append(containerIDs, cid)
+	}
+
+	pidsSorted := sort.IntSlice(pids)
+	pidsSorted.Sort()
+	for _, pid := range pidsSorted {
+		data.Processes = append(data.Processes, processes[int32(pid)])
+	}
+
+	containerIDsSorted := sort.StringSlice(containerIDs)
+	containerIDsSorted.Sort()
+	for _, cid := range containerIDsSorted {
+		data.Containers = append(data.Containers, containers[cid])
+	}
+
+	templates := []string{
+		processesTemplate,
+		containersTemplate,
+	}
+
+	for idx, name := range templates {
+		t := template.Must(template.New("process-" + strconv.Itoa(idx)).Funcs(fnMap).Parse(name))
+		err := t.Execute(w, data)
+		if err != nil {
+			return err
+		}
+		_, _ = w.Write(([]byte)("\n"))
+	}
+	return nil
+}
+
+func humanFormatRealTimeProcess(msgs []model.MessageBody, w io.Writer) error {
+	var data struct {
+		ProcessStats   []*model.ProcessStat
+		ContainerStats []*model.ContainerStat
+		Hostname       string
+		CPUCount       int
+		Memory         uint64
+	}
+
+	var (
+		processStats   = map[int32]*model.ProcessStat{}
+		containerStats = map[string]*model.ContainerStat{}
+		pids           []int
+		containerIDs   []string
+	)
+
+	for _, m := range msgs {
+		proc := m.(*model.CollectorRealTime)
+		data.Hostname = proc.HostName
+		data.Memory = uint64(proc.TotalMemory)
+		data.CPUCount = int(proc.NumCpus)
+		for _, p := range proc.Stats {
+			processStats[p.Pid] = p
+			pids = append(pids, int(p.Pid))
+		}
+
+		for _, c := range proc.ContainerStats {
+			containerStats[c.Id] = c
+		}
+	}
+
+	for cid := range containerStats {
+		containerIDs = append(containerIDs, cid)
+	}
+
+	pidsSorted := sort.IntSlice(pids)
+	pidsSorted.Sort()
+	for _, pid := range pidsSorted {
+		data.ProcessStats = append(data.ProcessStats, processStats[int32(pid)])
+	}
+
+	containerIDsSorted := sort.StringSlice(containerIDs)
+	containerIDsSorted.Sort()
+	for _, cid := range containerIDsSorted {
+		data.ContainerStats = append(data.ContainerStats, containerStats[cid])
+	}
+
+	templates := []string{
+		rtProcessesTemplate,
+		rtContainersTemplate,
+	}
+
+	for idx, name := range templates {
+		t := template.Must(template.New("process-" + strconv.Itoa(idx)).Funcs(fnMap).Parse(name))
+		err := t.Execute(w, data)
+		if err != nil {
+			return err
+		}
+		_, _ = w.Write(([]byte)("\n"))
+	}
+	return nil
+}
+
+func humanFormatContainer(msgs []model.MessageBody, w io.Writer) error {
+	var data struct {
+		Containers []*model.Container
+		Hostname   string
+		CPUCount   int
+		Memory     uint64
+	}
+
+	var (
+		containers   = map[string]*model.Container{}
+		containerIDs []string
+	)
+
+	for _, m := range msgs {
+		cont := m.(*model.CollectorContainer)
```

Comment:
Nit: We might want to throw an error if the type can't be converted, since it's possible to pass a type other than CollectorContainer

Reply:
Added error handling here.